src/main_benchmark.cpp

```diff
@@ -218,8 +224,96 @@ void triggerInferenceInALoop(
     averageWholeLatency = std::accumulate(latenciesWhole.begin(), latenciesWhole.end(), 0) / (double(niterPerThread) * 1'000);
     averagePureLatency = std::accumulate(latenciesPure.begin(), latenciesPure.end(), 0) / (double(niterPerThread) * 1'000);
 }
+
+void triggerInferenceInALoop(
+    std::future<void>& startSignal,
+    std::promise<void>& readySignal,
+    const size_t niterPerThread,
+    size_t& wholeThreadTimeUs,
+    double& averageWholeLatency,
+    double& averagePureLatency,
+    OVMS_Server* server,
+    OVMS_InferenceRequest* request,
+    const std::string& inputName,
+    std::uint32_t elementsCount) {
+    OVMS_InferenceResponse* response{nullptr};
+    std::vector<uint64_t> latenciesWhole(niterPerThread);
+    std::vector<uint64_t> latenciesPure(niterPerThread);
+    readySignal.set_value();
+    startSignal.get();
+    auto workloadStart = std::chrono::high_resolution_clock::now();
+    size_t iter = niterPerThread;
+    while (iter-- > 0) {
+        float random = ((float) rand()) / (float) RAND_MAX;
```

Comment:
better approach would be to prepare vectors of various buffers and then in workload loop just select different buffer each time eg with modulo operator. Otherwise workload measures also allocating input data.